packages/tocco-util/src/form/formBuilder.js

```diff
@@ -136,18 +136,31 @@ export default (
 
   const createLayoutComponent = (field, type, key, traverser) => {
     if (type === layoutTypes.HORIZONTAL_BOX || type === layoutTypes.VERTICAL_BOX) {
-      const alignment = type === layoutTypes.HORIZONTAL_BOX ? 'horizontal' : 'vertical'
-      const children = traverser()
+      let output = traverser()
 
-      if (children == null || (Array.isArray(children) && children.every(e => e == null))) {
+      if (Array.isArray(output)) {
+        output = output.filter(el => el)
+      }
+
+      if (!output || (Array.isArray(output) && output.every(e => e == null))) {
         return null
       }
 
-      return (
-        <LayoutBox key={key} label={field.label} alignment={alignment}>
-          {children}
-        </LayoutBox>
-      )
+      if (field.label) {
+        output
+        = <Panel.Wrapper isFramed={true} isOpen={true}>
+            <Panel.Header><Typography.H4>{field.label}</Typography.H4></Panel.Header>
+            <Panel.Body>{output}</Panel.Body>
+          </Panel.Wrapper>
+      }
+
+      if (type === layoutTypes.HORIZONTAL_BOX) {
+        output = <Layout.Container key={key}>{output}</Layout.Container>
+      } else {
```

Comment:
maybe handle all layout types explicitly and warn if the layout type is unknown?

Reply:
see commit bd0fca475d99cce820e75330d2095f459cbf14a7